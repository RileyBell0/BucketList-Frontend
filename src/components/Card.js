import "../styles/card.css";
import React from "react";

function Card({ children, className, onClick }) {
  return (
    <div
      className={"card" + (className ? " " + className : "")}
      onClick={onClick}
    >
      {children}
    </div>
  );
}

export default Card;
