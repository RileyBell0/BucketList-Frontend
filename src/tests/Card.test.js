import renderer from "react-test-renderer";
import Card from "../components/Card";

it("Matches previous snapshot", () => {
  const component = renderer.create(<Card>Hello!</Card>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("Applies classname correctly", () => {
  const comp = renderer.create(
    <Card className={"testing-class"}>Content</Card>
  );
  expect(comp.toJSON().props.className).toContain("testing-class");
});

it("Functions properly with no classname input", () => {
  const comp = renderer.create(<Card>Content</Card>);
  expect(comp.toJSON().props.className).toBe("card");
});
