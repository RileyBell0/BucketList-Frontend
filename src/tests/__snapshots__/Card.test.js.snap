// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches previous snapshot 1`] = `
<div
  className="card"
>
  Hello!
</div>
`;
